pkg/controller/machine.go

```diff
@@ -102,12 +102,12 @@ type KubeconfigProvider interface {
 // MetricsCollection is a struct of all metrics used in
 // this controller.
 type MetricsCollection struct {
-	Machines            metrics.Gauge
-	Nodes               metrics.Gauge
-	Workers             metrics.Gauge
-	Errors              metrics.Counter
-	ControllerOperation metrics.Histogram
-	NodeJoinDuration    metrics.Histogram
+	Machines            prometheus.Gauge
```

Comment:
Was it intended to move them in this PR as well?

Reply:
This PR is the reason I did that change. The `syncHandler` crashes with an NPE if the metrics are not initialized.

I just put those changes in a dedicated PR to easen review, will rebase this one once the one got merged